client/my-sites/checkout/checkout/stripe-elements-payment-box.jsx

```diff
@@ -0,0 +1,216 @@
+			setStripeJs( window.Stripe( apiKey ) );
+			return;
+		}
+		debug( 'loading stripe.js...' );
+		loadScript( url, function( error ) {
+			if ( error ) {
+				debug( 'stripe.js script ' + error.src + ' failed to load.' );
+				return;
+			}
+			debug( 'stripe.js loaded!' );
+			setStripeJs( window.Stripe( apiKey ) );
+		} );
+	}, [ url, apiKey ] );
+	return stripeJs;
+}
+
+async function submitPaymentForm( { stripe, paymentDetails, cart, transaction } ) {
+	const stripePaymentMethod = await createStripePaymentMethod( stripe, paymentDetails );
+	return sendStripePaymentMethodToStore( {
+		cart,
+		transaction,
+		paymentDetails,
+		stripePaymentMethod,
+	} );
+}
+
+async function createStripePaymentMethod( stripe, paymentDetails ) {
+	debug( 'creating payment method...', paymentDetails );
+	const { paymentMethod, error } = await stripe.createPaymentMethod( 'card', {
+		billing_details: paymentDetails,
+	} );
+	debug( 'payment method creation complete', paymentMethod, error );
+	if ( error ) {
+		// Note that this is a promise rejection
+		throw new Error( error.message );
+	}
+	return paymentMethod;
+}
+
+async function sendStripePaymentMethodToStore( {
+	cart,
+	transaction,
+	paymentDetails,
+	stripePaymentMethod,
+} ) {
+	const dataForApi = {
+		stripePaymentMethod, // TODO: the API does not accept this yet
+		payment: Object.assign( {}, paymentDetails, {
+			paymentMethod: paymentMethodClassName( 'stripe' ),
+			successUrl: '/',
+			cancelUrl: '/',
+		} ),
+		cart,
+		domain_details: transaction.domainDetails,
+	};
+
+	return wpcom.undocumented().transactions( 'POST', dataForApi );
+}
+
+function StripeElementsForm( {
+	translate,
+	stripe,
+	cart,
+	transaction,
+	children,
+	presaleChatAvailable,
+} ) {
+	// TODO: allow disabling the payment button during processing or when not ready
+	const [ cardholderName, setCardholderName ] = useState( '' );
+	const onNameChange = event => setCardholderName( event.target.value );
+	const handleSubmit = event => {
+		event.preventDefault();
+		submitPaymentForm( {
+			stripe,
+			cart,
+			transaction,
+			paymentDetails: {
+				name: cardholderName,
+			},
+		} )
+			.then( result => {
+				if ( result.redirect_url ) {
+					notices.success(
+						translate( 'Redirecting you to the payment partner to complete the payment.' )
+					);
+					window.location.href = result.redirect_url;
+				}
+				// TODO: what if there is no redirect_url?
+			} )
+			.catch( error => {
+				const errorMessage =
+					error.message || translate( "We've encountered a problem. Please try again later." );
+				notices.error( errorMessage );
+			} );
+	};
+	const cardholderNameLabel = translate( 'Cardholder Name {{span}}(as written on card){{/span}}', {
+		comment: 'Cardholder name label on credit card form',
+		components: {
+			// eslint-disable-next-line wpcalypso/jsx-classname-namespace
+			span: <span className="credit-card-form-fields__explainer" />,
+		},
+	} );
+	const cardDetailsLabel = translate( 'Card Details' );
```

Comment:
Hi! I've found a possible matching string that has already been translated **20** times:
`translate( 'Edit Card Details', { context: 'Header text'} )` *ES Score: 7*
See 1 additional suggestion in the [PR translation status page](https://translate.wordpress.com/localci/status/automattic/wp-calypso/add/stripe-elements-checkout)

Help me improve these suggestions: react with :thumbsdown: if the suggestion doesn't make any sense, or with :thumbsup: if it's a particularly good one (even if not implemented).